### ISSUE TITLE
low success rate

### ISSUE THREAD

**Issue by jayyuci** [siyuhsu/vla-cache#5]
Excuse me, I conduct evaluation script on my computer, and achieve very low success rate. Is there anything wrong with my configurations?
I test on a 4090, linux server.
The success rate is 0, whether the --use_vla_ache is Ture or False

below is my evaluation script and evaluation results.

![Image](https://github.com/user-attachments/assets/b3aca6f8-d8d0-4209-8370-20161112b0d3)

![Image](https://github.com/user-attachments/assets/bd8174f9-c8ee-4482-89e9-362e51a055fd)

**Comment by siyuhsu** [comment]
It looks like the issue might be related to your Python environment not having the modified openvla version with VLA-Cache properly installed. 

Please try reinstalling it using the following commands:

```
conda activate openvla
cd src/openvla
pip install -e .
```

**Comment by jayyuci** [comment]
it seems that the problem stems from there, error message shows:
![Image](https://github.com/user-attachments/assets/0ef7713f-c9a8-461a-891f-20507bdd2434)
which stems mismatch from

![Image](https://github.com/user-attachments/assets/2cd67c64-a5e0-4242-9da9-c8bfde3791e7)
and

![Image](https://github.com/user-attachments/assets/44384767-72a5-4852-940c-9198a9e8ce24)

which result in that the VLA do nothing in each run.

**Comment by siyuhsu** [comment]
From your screenshots, it looks like the `vla-cache` modification did **not take effect**, which is likely why the evaluation is not running correctly. Please follow these steps to ensure everything is set up properly:

1. **Reinstall the modified `openvla` module**  
   Make sure you’re in the correct environment and install the local version with `VLA-Cache` modifications:

   ```bash
   conda activate openvla
   cd src/openvla
   pip install -e .
   ```

2. **Download the correct pretrained model to the project directory**  
   This ensures the weights are correctly loaded from the modified version:

   ```bash
   python vla_cache_scripts/download_model_local.py \
     --model_id openvla/openvla-7b-finetuned-libero-spatial
   ```
   The checkpoints will be saved to `src/openvla/checkpoinsts`

3. **Set the `--pretrained_checkpoint` path correctly in your evaluation script**  
   Make sure the weights are loaded from the local path:

   ```bash
   --pretrained_checkpoint checkpoints/openvla-7b-finetuned-libero-spatial
   ```

If the weights are not loaded from the local `checkpoints` directory, the system may fallback to using the unmodified `OpenVLAForActionPrediction` class, causing `VLA-Cache` to have no effect.

Let me know if this resolves the issue!

**Comment by jayyuci** [comment]
> From your screenshots, it looks like the `vla-cache` modification did **not take effect**, which is likely why the evaluation is not running correctly. Please follow these steps to ensure everything is set up properly:
> 
> 1. **Reinstall the modified `openvla` module**
>    Make sure you’re in the correct environment and install the local version with `VLA-Cache` modifications:
>    conda activate openvla
>    cd src/openvla
>    pip install -e .
> 2. **Download the correct pretrained model to the project directory**
>    This ensures the weights are correctly loaded from the modified version:
>    python vla_cache_scripts/download_model_local.py \
>      --model_id openvla/openvla-7b-finetuned-libero-spatial
>        
>          
>        
>    
>          
>        
>    
>        
>      
>    The checkpoints will be saved to `src/openvla/checkpoinsts`
> 3. **Set the `--pretrained_checkpoint` path correctly in your evaluation script**
>    Make sure the weights are loaded from the local path:
>    --pretrained_checkpoint checkpoints/openvla-7b-finetuned-libero-spatial
> 
> If the weights are not loaded from the local `checkpoints` directory, the system may fallback to using the unmodified `OpenVLAForActionPrediction` class, causing `VLA-Cache` to have no effect.
> 
> If the issue still persists after completing the steps above, please **share your full evaluation command and a complete screenshot or log of the output** so we can help you debug further.

Thank you! The problem lies in I directly use the checkpoint I downloaded before, now the program can run fluently.
I just rerun the command in openvla, of the **object task**, **without vla-cache**
however, I didn't manage to achieve the result in the article. Is there anything wrong with my settings?
the command and result are as follows:

![Image](https://github.com/user-attachments/assets/cbc6b15a-20ea-4635-94a0-e02c3b46732e)

![Image](https://github.com/user-attachments/assets/a7510364-5437-4b8d-b1da-6306a8f0be5a)

and the  inference cuda time is significantly different. I wonder whether it is because of the matter of GPU. I conducted the evaluation task on 4090.

![Image](https://github.com/user-attachments/assets/df4ef5aa-0b7b-4f41-93eb-4903e7681651)

**Comment by siyuhsu** [comment]
The large gap between **current** and **average latency** does seem abnormal. This might indicate some issue with how the environment or code is set up.

Could you please try the following:

- **Re-clone** the `vla-cache` repository
- Set up the environment following the instruction step by step **without making any modifications**
- Run the evaluation script again with the default settings

Let me know how it goes!

**Comment by jayyuci** [comment]
> The large gap between **current** and **average latency** does seem abnormal. This might indicate some issue with how the environment or code is set up.
> 
> Could you please try the following:
> 
> * **Re-clone** the `vla-cache` repository
> * Set up the environment following the instruction step by step **without making any modifications**
> * Run the evaluation script again with the default settings
> 
> Let me know how it goes!

In fact, I totally follow the instruction the repo provides, starting from **Re-clone** the `vla-cache` repository
the difference may lie in the installation of the libero env, which I copy from the openvla repo

![Image](https://github.com/user-attachments/assets/b8f8ae5d-238e-4ee2-9608-d000ccb0c90b)

I am currently trying to reproduce the **vla-cache mode** of the openvla, object. the result is as follows:

![Image](https://github.com/user-attachments/assets/97b7c974-1215-48f3-9a2f-d6039bfcb5c7)

**Comment by jayyuci** [comment]
In our reimplentation, we found three important hypterparameters, which have great impact on the accuracy

**top_k in the task_relevant_selection**
 ![Image](https://github.com/user-attachments/assets/7eb2a550-3d63-4c50-84df-05abf997c85c)

**sim_threshold** and **top_k** in the **find_static_patches**

![Image](https://github.com/user-attachments/assets/28c64b0d-af3b-408e-b2fd-47104d8e01c4)

in the code, we find that it is different from the paper, we set these hyperparameter the same as that in the paper
(find_static_patches:
top_k=100,
sim_threshold=0.996
)
task_relevant_selection
top_k = 120
**the paper set a threshold=0.5**, but there is no such parameter

but I still cannot reproduce the result in Libero Object

![Image](https://github.com/user-attachments/assets/23c51575-531a-44da-9584-c07af951992a)

**Comment by siyuhsu** [comment]
You need to start by reproducing the baseline results first. In our recent evaluations on a single 4090 GPU, the baseline success rate for the Libero Object task reached 85.2%. If your baseline result is significantly lower, there might be some underlying issues or bugs that need to be resolved first.

Additionally, it’s important to evaluate all subtasks. In our experiments, we also observed that the success rate of VLA-Cache can be lower at the beginning, but it gradually improves and eventually approaches the baseline level after testing all tasks.

Regarding the hyperparameters you mentioned, they are indeed important. The implementation in the code directly uses top-k selection to retrieve task-relevant tokens, which is a more efficient approach as it avoids additional computation overhead. This method has a similar effect to the threshold-based $\tau%$ selection described in the paper. We will clarify and revise this point in the updated version of the paper.

**Comment by jayyuci** [comment]
> You need to start by reproducing the baseline results first. In our recent evaluations on a single 4090 GPU, the baseline success rate for the Libero Object task reached 85.2%. If your baseline result is significantly lower, there might be some underlying issues or bugs that need to be resolved first.
> 
> Additionally, it’s important to evaluate all subtasks. In our experiments, we also observed that the success rate of VLA-Cache can be lower at the beginning, but it gradually improves and eventually approaches the baseline level after testing all tasks.
> 
> Regarding the hyperparameters you mentioned, they are indeed important. The implementation in the code directly uses top-k selection to retrieve task-relevant tokens, which is a more efficient approach as it avoids additional computation overhead. This method has a similar effect to the threshold-based τ selection described in the paper. We will clarify and revise this point in the updated version of the paper.


Thank you for your patient response. We reinplement the **object** task with **vla-cache** mode, however, the result is not as expected

![Image](https://github.com/user-attachments/assets/4c362925-e27c-4b0e-9832-236a74c626be)

as for the hyperparameter setting, we set the 
- top_k = 100, sim_threshold=0.996 in find_static_patches
- top_k = 120 in task_relevant_selection

is there anything wrong with our settings?

**Comment by siyuhsu** [comment]
We reconfigured the environment from scratch inside a Docker container yesterday, strictly following the instructions provided in the repo. We then reproduced the VLA-Cache implementation and ran the LIBERO Spatial benchmark on a single 4090 GPU.

Below are the final evaluation results along with the corresponding logs for reference:

[EVAL-libero_spatial-openvla-2025_06_28-23_04_08_baseline.txt](https://github.com/user-attachments/files/20965214/EVAL-libero_spatial-openvla-2025_06_28-23_04_08_baseline.txt)
[EVAL-libero_spatial-openvla-2025_06_28-23_06_02_vla-cache.txt](https://github.com/user-attachments/files/20965215/EVAL-libero_spatial-openvla-2025_06_28-23_06_02_vla-cache.txt)


It seems there may be an issue with your current environment. We recommend trying to reconfigure the environment from scratch—preferably on a different machine or using Docker—to ensure a consistent setup and avoid potential conflicts.

**Comment by jayyuci** [comment]
Thank you for your feedback. With you help, I did successfully reproduce the result of the **spatial** task. However, I didn't manage to reproduce the result on the other 3 tasks. I append the result below.

[EVAL-libero_10-openvla-2025_07_01-05_35_33.txt](https://github.com/user-attachments/files/21009715/EVAL-libero_10-openvla-2025_07_01-05_35_33.txt)
[EVAL-libero_goal-openvla-2025_07_01-01_25_25.txt](https://github.com/user-attachments/files/21009718/EVAL-libero_goal-openvla-2025_07_01-01_25_25.txt)
[EVAL-libero_object-openvla-2025_06_30-20_24_16.txt](https://github.com/user-attachments/files/21009717/EVAL-libero_object-openvla-2025_06_30-20_24_16.txt)
[EVAL-libero_spatial-openvla-2025_06_30-11_26_54.txt](https://github.com/user-attachments/files/21009716/EVAL-libero_spatial-openvla-2025_06_30-11_26_54.txt)

**Comment by ChuhangLiu2002** [comment]
> > Have you reproduced the result of **Object** on the baseline setting?
> 
> Yeah, I just try to reproduce the outcome on the **object** baseline setting, and I cannot reproduce the expecting result either. The result is appended.
> 
> [EVAL-libero_object-openvla-2025_07_02-09_54_49.txt](https://github.com/user-attachments/files/21013711/EVAL-libero_object-openvla-2025_07_02-09_54_49.txt)
> 
> I check the openvla repo, following the original setting, I added the **center_crop** param, the result didn' change either, only reach 63.8(vla-cache mode)

Thanks for your question, have you solved this problem? I've encountered the same issue.

**Comment by Wpq-ai** [comment]
> > From your screenshots, it looks like the `vla-cache` modification did **not take effect**, which is likely why the evaluation is not running correctly. Please follow these steps to ensure everything is set up properly:
> > 
> > 1. **Reinstall the modified `openvla` module**
> >    Make sure you’re in the correct environment and install the local version with `VLA-Cache` modifications:
> >    conda activate openvla
> >    cd src/openvla
> >    pip install -e .
> > 2. **Download the correct pretrained model to the project directory**
> >    This ensures the weights are correctly loaded from the modified version:
> >    python vla_cache_scripts/download_model_local.py 
> >    --model_id openvla/openvla-7b-finetuned-libero-spatial
> >    The checkpoints will be saved to `src/openvla/checkpoinsts`
> > 3. **Set the `--pretrained_checkpoint` path correctly in your evaluation script**
> >    Make sure the weights are loaded from the local path:
> >    --pretrained_checkpoint checkpoints/openvla-7b-finetuned-libero-spatial
> > 
> > If the weights are not loaded from the local `checkpoints` directory, the system may fallback to using the unmodified `OpenVLAForActionPrediction` class, causing `VLA-Cache` to have no effect.
> > If the issue still persists after completing the steps above, please **share your full evaluation command and a complete screenshot or log of the output** so we can help you debug further.
> 
> Thank you! The problem lies in I directly use the checkpoint I downloaded before, now the program can run fluently. I just rerun the command in openvla, of the **object task**, **without vla-cache** however, I didn't manage to achieve the result in the article. Is there anything wrong with my settings? the command and result are as follows:
> 
> ![Image](https://github.com/user-attachments/assets/cbc6b15a-20ea-4635-94a0-e02c3b46732e)
> 
> ![Image](https://github.com/user-attachments/assets/a7510364-5437-4b8d-b1da-6306a8f0be5a)
> 
> and the inference cuda time is significantly different. I wonder whether it is because of the matter of GPU. I conducted the evaluation task on 4090.
> 
> ![Image](https://github.com/user-attachments/assets/df4ef5aa-0b7b-4f41-93eb-4903e7681651)

I met this problem

> > From your screenshots, it looks like the `vla-cache` modification did **not take effect**, which is likely why the evaluation is not running correctly. Please follow these steps to ensure everything is set up properly:
> > 
> > 1. **Reinstall the modified `openvla` module**
> >    Make sure you’re in the correct environment and install the local version with `VLA-Cache` modifications:
> >    conda activate openvla
> >    cd src/openvla
> >    pip install -e .
> > 2. **Download the correct pretrained model to the project directory**
> >    This ensures the weights are correctly loaded from the modified version:
> >    python vla_cache_scripts/download_model_local.py 
> >    --model_id openvla/openvla-7b-finetuned-libero-spatial
> >    The checkpoints will be saved to `src/openvla/checkpoinsts`
> > 3. **Set the `--pretrained_checkpoint` path correctly in your evaluation script**
> >    Make sure the weights are loaded from the local path:
> >    --pretrained_checkpoint checkpoints/openvla-7b-finetuned-libero-spatial
> > 
> > If the weights are not loaded from the local `checkpoints` directory, the system may fallback to using the unmodified `OpenVLAForActionPrediction` class, causing `VLA-Cache` to have no effect.
> > If the issue still persists after completing the steps above, please **share your full evaluation command and a complete screenshot or log of the output** so we can help you debug further.
> 
> Thank you! The problem lies in I directly use the checkpoint I downloaded before, now the program can run fluently. I just rerun the command in openvla, of the **object task**, **without vla-cache** however, I didn't manage to achieve the result in the article. Is there anything wrong with my settings? the command and result are as follows:
> 
> ![Image](https://github.com/user-attachments/assets/cbc6b15a-20ea-4635-94a0-e02c3b46732e)
> 
> ![Image](https://github.com/user-attachments/assets/a7510364-5437-4b8d-b1da-6306a8f0be5a)
> 
> and the inference cuda time is significantly different. I wonder whether it is because of the matter of GPU. I conducted the evaluation task on 4090.
> 
> ![Image](https://github.com/user-attachments/assets/df4ef5aa-0b7b-4f41-93eb-4903e7681651)

I met this problem as well, Can you tell me how you solve this problem in detail?
